docs/pages/kubernetes-access/controls.mdx

```diff
@@ -639,16 +639,25 @@ these values to a list of RBAC principals it will use to write impersonation
 headers later on.
 
 Next, the Kubernetes Service checks each of the user's roles for `spec.deny`
-conditions. If one role's `spec.deny.kubernetes_labels` or
-`spec.deny.kubernetes_resources` fields match the user's request, the Kubernetes
-Service looks up the role's `spec.deny.kubernetes_groups` and
-`spec.deny.kubernetes_users` fields. It removes each of these values from the
-list of users and groups it created earlier, denying the user access to these
-RBAC principals.
+conditions that might apply to the request:
+
+- For `spec.deny.kubernetes_labels`, if any labels match the cluster's labels 
+  for the request, the Service denies the request before it is made to the 
+  Kubernetes API. 
+- For `spec.deny.kubernetes_resources`, if the resource matches a `deny` condition, 
```

Comment:
In this case, the behavior is that you can't do the request. We won't change the `kubernetes_groups` or users.
If you have `kubernetes_users|groups` deny conditions, you can't use them at all